### PR TITLE
deprecate Cyclic.ZModulo.ZModulo

### PR DESCRIPTION
**Pull request by andres-erbsen**
The string `ZModulo` does not appear anywhere in the CI; in this repo it only appears in the [stdlib index](https://github.com/coq/coq/blob/790030f359136a2a31124236587bac53db5fdfcf/doc/stdlib/index-list.html.template#L282).

My only encounters with this library have been when it is mistaken for a general implementation of integers modulo another integer, much earlier by myself and recently by others.

I think having a sanity-check for `CyclicAxioms` is valuable, but can be accomplished under a less catchy name or even in the test suite.

This PR is intended to be merged in addition to https://github.com/coq/coq/pull/16913

- [x] Added **changelog**.

### CODE REVIEW

**Comment by JasonGross** [comment]
What about renaming the file to `ZModuloPow2` or something like that, and than make a new file `ZModulo.v` which does `#[deprecated(...)] Include ZModuloPow2.` (does `#[deprecated]` work well on `Include`?)

**Comment by andres-erbsen** [comment]
`Include` does not support `deprecated`. There might be still me an automated mechanism; I followed `theories/Numbers/Cyclic/Int31/Cyclic31.v` which was depreceted in 8.10.

**Comment by andres-erbsen** [comment]
@letouzey  any thoughts? I'm thinking a good outcome would be eventually moving this file to `test-suite`.

> commit 1b5247722681a8e22608b1c5b017c6faf2b3c7fc
> Author: Pierre Letouzey <pierre.letouzey@inria.fr>
> Date:   Sat May 17 23:23:15 2008 +0000
>
>   ZModulo: Z viewed modulo 2^digits implements CyclicAxioms
>  
>   This isn't useful for BigN et BigZ, but it can't hurt; and moreover
>   it's a simple way to understand CyclicAxioms. Next step: proving that
>   Int31 is also an implementation of CyclicAxiom.

**Comment by Zimmi48** [comment]
Moving it to the test suite sounds reasonable. The changelog could explain why it's deprecated. @JasonGross Will you be an assignee for this PR?

**Review by JasonGross — CHANGES_REQUESTED** [review]
Following @Zimmi48 's suggestion, could you add the justification from the PR comment to the changelog entry, to and then I'll merge within about 24 hours?

**Review by JasonGross — APPROVED** [review]

**Comment by JasonGross** [comment]
I will merge once the doc job builds (ensuring that the changelog has proper syntax)

File: doc/changelog/11-standard-library/16914-deprecate-ZModulo.rst
Comment (JasonGross) [inline doc/changelog/11-standard-library/16914-deprecate-ZModulo.rst:1]:
```suggestion
- **Deprecated:** :g:`Cyclic.ZModulo.ZModulo` because there have been no
```

Comment (JasonGross) [reply doc/changelog/11-standard-library/16914-deprecate-ZModulo.rst:1]: Linter fails from extra trailing space

Comment (Zimmi48) [reply doc/changelog/11-standard-library/16914-deprecate-ZModulo.rst:1]: @JasonGross Note that it would be OK for you to commit this suggestion before merging. That being said, if @andres-erbsen does it, this could be an opportunity to squash the commits that amend each other.

**Comment by JasonGross** [comment]
@coqbot merge now

**Comment by JasonGross** [comment]
@coqbot merge now

**Pull request merged**
